### ISSUE TITLE
PendingDeprecationWarning: generator 'pytest_pycollect_makeitem' raised StopIteration

### ISSUE THREAD

**Issue by 5j9** [pytest-dev/pluggy#38]
Hi,

I'm getting lots of these warnings when running some of the [pywikibot ](https://github.com/wikimedia/pywikibot-core) tests using pytest and python 3.5.

```lang=python
WARNING: C:\Users\a\AppData\Local\Programs\Python\Python35\lib\site-packages\_pytest\vendored_packages\pluggy.py:250: PendingDeprecationWarning: generator 'pytest_pycollect_makeitem' raised StopIteration
  wrap_controller.send(call_outcome)
```

See: https://travis-ci.org/wikimedia/pywikibot-core/jobs/167265098#L493

**Comment by tgoodlet** [comment]
@RonnyPfannschmidt @5j9  this is due to [PEP 479](https://docs.python.org/3/whatsnew/3.5.html#pep-479-change-stopiteration-handling-inside-generators) which is in full [here](https://www.python.org/dev/peps/pep-0479/).

So it's the underlying hookwrapper which are triggering this warning not `pluggy` code.

In this case the default implementation of [`pytest_pycollect_makeitem`](https://github.com/pytest-dev/pytest/blob/3c81f83602e5fa38cb6187dfc7b131ed88bf8b7d/_pytest/python.py#L177) is raising `StopIteration` (deprecated in py3.6) instead of using `return`.

However, in order to support py3.6 we need to change `_wrapped_call()` to except `RuntimeError`s alongside `StopIteration`s.

@RonnyPfannschmidt I believe and issue with `pytest` should be made?

**Comment by RonnyPfannschmidt** [comment]
Correct, thanks for investigation

**Comment by tgoodlet** [comment]
@RonnyPfannschmidt pretty sure this can be closed now?

**Comment by nicoddemus** [comment]
Yep, thanks for the ping!

**Comment by nicoddemus** [comment]
Fixed in pytest-dev/pytest#2226
